net-misc/FORT-validator/FORT-validator-1.2.0.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools fcaps
+
+DESCRIPTION="FORT validator is an open source RPKI validator."
+HOMEPAGE="https://fortproject.net/validator?2"
+SRC_URI="https://github.com/NICMx/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+MY_PN="fort"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="caps libressl"
+
+DEPEND="
+	acct-group/fort
+	acct-user/fort
+	caps? ( sys-libs/libcap )
+	dev-libs/jansson
+	libressl?	( dev-libs/libressl:0= )
+	!libressl?	( dev-libs/openssl:0= )
+	net-misc/rsync
```

Comment:
How is rsync used?

Reply:
It’s used to fill your cache by polling ROAs from repos (usually maintained by RIRs): https://tools.ietf.org/html/rfc8210#section-2
It looks like this on htop:
```
  36232 fort       20   0  513M 32412  7116 S 77.9  1.6  0:44.68 ├─ /usr/bin/fort --configuration-file /etc/fort/config.json
  36290 fort       20   0  212M  2812  2616 S  0.0  0.1  0:00.00 │  ├─ rsync --recursive --delete --times --contimeout=20 --timeout=15 rsync://rpki.admin.freerangecloud.com/repo /var/cache/fort/repository/rpki.admin.freerangecloud.com/re
  36287 fort       20   0  259M  4540  2560 S  0.0  0.2  0:00.08 │  ├─ rsync --recursive --delete --times --contimeout=20 --timeout=15 rsync://rpki.cnnic.cn/rpki /var/cache/fort/repository/rpki.cnnic.cn/rpki
  36288 fort       20   0  213M  1772  1380 S 12.0  0.1  0:00.21 │  │  └─ rsync --recursive --delete --times --contimeout=20 --timeout=15 rsync://rpki.cnnic.cn/rpki /var/cache/fort/repository/rpki.cnnic.cn/rpki
```

And like this on tcpdump:
```
ares ~ # tcpdump -i enp0s18 port rsync or port https
dropped privs to tcpdump
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on enp0s18, link-type EN10MB (Ethernet), capture size 262144 bytes
09:39:39.663546 IP6 ares.swordarmor.fr.45302 > rpki.apnic.net.rsync: Flags [S], seq 2363839765, win 64800, options [mss 1440,sackOK,TS val 4229636855 ecr 0,nop,wscale 7], length 0
09:39:39.666503 IP6 ares.swordarmor.fr.48410 > rpki.afrinic.net.rsync: Flags [S], seq 1800594157, win 64800, options [mss 1440,sackOK,TS val 3252885817 ecr 0,nop,wscale 7], length 0
09:39:39.781691 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [S], seq 1015639874, win 64800, options [mss 1440,sackOK,TS val 3761607054 ecr 0,nop,wscale 7], length 0
09:39:39.807567 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [S.], seq 3144277359, ack 1015639875, win 23040, options [mss 1440,nop,wscale 9,sackOK,TS val 3769371648 ecr 3761607054], length 0
09:39:39.807696 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [.], ack 1, win 507, options [nop,nop,TS val 3761607080 ecr 3769371648], length 0
09:39:39.807992 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [P.], seq 1:15, ack 1, win 507, options [nop,nop,TS val 3761607080 ecr 3769371648], length 14
09:39:39.833561 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [.], ack 15, win 45, options [nop,nop,TS val 3769371674 ecr 3761607080], length 0
09:39:39.840943 IP6 rpki.afrinic.net.rsync > ares.swordarmor.fr.48410: Flags [S.], seq 1682971404, ack 1800594158, win 14280, options [mss 1440,sackOK,TS val 42721250 ecr 3252885817,nop,wscale 7], length 0
09:39:39.841048 IP6 ares.swordarmor.fr.48410 > rpki.afrinic.net.rsync: Flags [.], ack 1, win 507, options [nop,nop,TS val 3252885991 ecr 42721250], length 0
09:39:39.841340 IP6 ares.swordarmor.fr.48410 > rpki.afrinic.net.rsync: Flags [P.], seq 1:15, ack 1, win 507, options [nop,nop,TS val 3252885992 ecr 42721250], length 14
09:39:39.866332 IP6 ares.swordarmor.fr.38120 > rpki.arin.net.rsync: Flags [S], seq 1406944681, win 64800, options [mss 1440,sackOK,TS val 3773973951 ecr 0,nop,wscale 7], length 0
09:39:39.880940 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [P.], seq 1:15, ack 15, win 45, options [nop,nop,TS val 3769371721 ecr 3761607080], length 14
09:39:39.881024 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [.], ack 15, win 507, options [nop,nop,TS val 3761607153 ecr 3769371721], length 0
09:39:39.881236 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [P.], seq 15:164, ack 15, win 45, options [nop,nop,TS val 3769371722 ecr 3761607080], length 149
09:39:39.881279 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [.], ack 164, win 506, options [nop,nop,TS val 3761607154 ecr 3769371722], length 0
09:39:39.881469 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [P.], seq 15:18, ack 164, win 506, options [nop,nop,TS val 3761607154 ecr 3769371722], length 3
09:39:39.906975 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [.], ack 18, win 45, options [nop,nop,TS val 3769371747 ecr 3761607154], length 0
09:39:39.907819 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [P.], seq 164:176, ack 18, win 45, options [nop,nop,TS val 3769371748 ecr 3761607154], length 12
09:39:39.908215 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [P.], seq 18:26, ack 176, win 506, options [nop,nop,TS val 3761607181 ecr 3769371748], length 8
09:39:39.933748 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [.], ack 26, win 45, options [nop,nop,TS val 3769371774 ecr 3761607181], length 0
09:39:39.933839 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [P.], seq 26:82, ack 176, win 506, options [nop,nop,TS val 3761607206 ecr 3769371774], length 56
09:39:39.959183 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [.], ack 82, win 45, options [nop,nop,TS val 3769371800 ecr 3761607206], length 0
09:39:39.960323 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [P.], seq 176:177, ack 82, win 45, options [nop,nop,TS val 3769371800 ecr 3761607206], length 1
09:39:39.960871 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [P.], seq 177:189, ack 82, win 45, options [nop,nop,TS val 3769371801 ecr 3761607206], length 12
09:39:39.961177 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [.], ack 189, win 506, options [nop,nop,TS val 3761607234 ecr 3769371800], length 0
09:39:39.961269 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [P.], seq 82:90, ack 189, win 506, options [nop,nop,TS val 3761607234 ecr 3769371800], length 8
09:39:39.961441 IP6 rpki.apnic.net.rsync > ares.swordarmor.fr.45302: Flags [S.], seq 78177307, ack 2363839766, win 22080, options [mss 1380,nop,wscale 9,sackOK,TS val 3368014890 ecr 4229636855], length 0
09:39:39.961576 IP6 ares.swordarmor.fr.45302 > rpki.apnic.net.rsync: Flags [.], ack 1, win 507, options [nop,nop,TS val 4229637153 ecr 3368014890], length 0
09:39:39.961763 IP6 ares.swordarmor.fr.45302 > rpki.apnic.net.rsync: Flags [P.], seq 1:15, ack 1, win 507, options [nop,nop,TS val 4229637153 ecr 3368014890], length 14
09:39:39.986895 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [.], ack 90, win 45, options [nop,nop,TS val 3769371827 ecr 3761607234], length 0
09:39:39.987626 IP6 rpki.ripe.net.rsync > ares.swordarmor.fr.51780: Flags [P.], seq 189:228, ack 90, win 45, options [nop,nop,TS val 3769371828 ecr 3761607234], length 39
09:39:39.988670 IP6 ares.swordarmor.fr.51780 > rpki.ripe.net.rsync: Flags [P.], seq 90:114, ack 228, win 506, options [nop,nop,TS val 3761607261 ecr 3769371828], length 24
^C09:39:40.581727 IP6 2600:9000:2171:d600:15:fa9:dc00:93a1.https > ares.swordarmor.fr.33314: Flags [.], seq 4258418402:4258432898, ack 2679983636, win 103, options [nop,nop,TS val 12320409 ecr 3397702352], length 14496

33 packets captured
19446 packets received by filter
19382 packets dropped by kernel
```

Reply:
Ook, it's used at runtime

Reply:
Yes indeed, corrected :)